### ISSUE TITLE
Updated master this morning, unable to compile (for linux nor win64 via mxe)

### ISSUE THREAD

**Issue by ProfoundDarkness** [CleverRaven/Cataclysm-DDA#51722]
### Describe the bug

After pulling the latest master this morning I am left unable to compile.  Both win64 cross compile with MXE and Linux native x64 fail to compile.  Seems like it might be translation related from the error.

### Steps To Reproduce

1. Pull latest master (My last pull is at e6a647a84285351f14033b41256eac9299caff2f )
2. Run make...

### Expected behavior

Successful compile, at least for Linux (I periodically have trouble cross compiling with MXE)

### Versions and configuration

- OS: Ubuntu 20.04.3 LTS
- Version: 0.F-2774-ge6a647a842

### Additional context

I'm going to concentrate on Linux native builds for now.

My current Linux compile line is:
make -j$((`nproc`+0)) NATIVE=linux64 CCACHE= RELEASE=1 SDL=1 TILES=1 SOUND=1 LTO=1 LOCALIZE= LINTJSON= ASTYLE= RUNTESTS= LANGUAGES=en

I didn't use ccache on that compile because I wanted to be sure it wasn't causing problems... I also did a 'make clean' prior.

The output from build is lengthy (as is the error message) so I've attached it as a file.
[buildinfo.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/7192614/buildinfo.txt)

**Comment by casswedson** [comment]
Should be fixed by #51718

**Comment by ProfoundDarkness** [comment]
Thanks, noticed *right* after posting this.
Testing builds now, will close on success.  Linux builds take a while, windows cross compile is usually quick (and I'm using ccache for both).

**Comment by ProfoundDarkness** [comment]
Still failing to build for Linux, new error, new build log attached.  This log is without ccache right after a make clean.  So same line as in initial issue.
[buildinfo.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/7192677/buildinfo.txt)

**Comment by ProfoundDarkness** [comment]
Noticed #51720 and thought I'd try building Linux with LOCALIZE=1.  That build succeeded but I got a whole mess of trouble trying to build for windows.

I went ahead and built for windows using the following line:
make -j$((`nproc`+0)) CROSS="${PLATFORM_64}" CCACHE=1 RELEASE=1 SDL=1 TILES=1 SOUND=1 LTO= LOCALIZE=1 LINTJSON= ASTYLE= RUNTESTS= PCH= LANGUAGES=en

I'm getting errors every time though if I just run the compile again it errors on a different file each time.  Something like this has happened before but not to this extent.  Now every couple of cpp files and in the end the build never succeeds.
[error_ex.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/7192940/error_ex.txt)

Once #51720 is merged I'll make sure Linux builds then try cross compile to Windows with the usual unset LOCALIZE flag which *was* working.


Aside: I don't normally set that LOCALIZE flag due to an error compiling .mo files...
https://github.com/CleverRaven/Cataclysm-DDA/blob/8ba81b4abbd3ce3e9be94b1440ed80f24befdb9d/lang/compile_mo.sh#L29
The double = in the test throws an error and doesn't succeed for me.

**Comment by BrettDong** [comment]
The minimum supported GCC version is 7.5: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/COMPILING/COMPILER_SUPPORT.md

The error log suggests you are using GCC 5.5.

**Comment by ProfoundDarkness** [comment]
I noticed the PR got merged so here goes tests...

For Linux (btw gcc ver 9.3.0) with the line: make -j$((`nproc`+0)) NATIVE=linux64 CCACHE=1 RELEASE=1 SDL=1 TILES=1 SOUND=1 LTO=1 PCH=1 LOCALIZE= LINTJSON= ASTYLE= RUNTESTS= LANGUAGES=en
Success!

As for windows...  sorry I didn't catch that... when I've tried to update MXE (even just starting from scratch following instructions) the builds it produces have major graphical errors.  Most recently (a couple of months at most) no foreground text but background colors.  So the start screen is mostly blank except for which item (ie load) is highlighted in which case just the background color is visible.  This is true with all text in game on those builds.

Anyway still doesn't build, not sure what to do to get windows cross compile up and running but that's my problem.  Anyway with the Linux builds working I'll go ahead and close this up, thanks.
